src/web-components/vf-stave.js

```diff
@@ -0,0 +1,206 @@
+// ## Description
+// 
+// This file implements `vf-stave`, the web component that resembles 
+// the `Stave` element. One `vf-stave` can have multiple `vf-voice` children. 
+// `vf-stave` is responsible for keeping track of its children and creating 
+// the voices once they finish generating their notes. 
+// Once the voices are created, `vf-stave` dispatches an event to its parent
+// `vf-system` to signal that it's ready to be created and added to the system. 
+
+import './vf-score';
+
+export class VFStave extends HTMLElement {
+
+  /**
+   * The Vex.Flow.Factory instance to use.
+   * @type {Vex.Flow.Factory}
+   * @private
+   */
+  _vf;
+
+  /**
+   * The Vex.Flow.Registry instance to use when registering elements.
+   * @type {Vex.Flow.Registry}
+   * @private
+   */
+  _registry;
+
+  /**
+   * The voices that this vf-stave has. 
+   * @type {[Vex.Flow.Voice]}
+   */
+  voices = [];
+
+  /**
+   * The beams for the voices that this vf-stave has. 
+   * @type {[Vex.Flow.Beam]}
+   */
+  beams = [];
+
+  constructor() {
+    super();
+
+    this.attachShadow({ mode:'open' });
+    this.shadowRoot.innerHTML = `<slot></slot>`;
+
+    // The 'notesCreated' event is dispatched by a vf-voice when it has 
+    // finished generating its notes. vf-stave listens to this event so that it 
+    // can get that vf-voice's notes and generate a Voice from it. 
+    this.addEventListener('notesCreated', this.addVoice);
+
+    // The 'vfVoiceReady' event is dispatched by a vf-voice when it's added to 
+    // the DOM. vf-stave listens to this event so that it can set the vf-voice's
+    // EasyScore instance, since a single EasyScore instance is shared by a 
+    // vf-stave and all its children. 
+    this.addEventListener('vfVoiceReady', this.setScore);
+  }
+
+  connectedCallback() {
+    this.clef = this.getAttribute('clef');
+    this.timeSig = this.getAttribute('timeSig');
+    this.keySig = this.getAttribute('keySig');
+
+    const vfStaveReadyEvent = new CustomEvent('vfStaveReady', { bubbles: true });
```

Comment:
You may consider making the string `'vfStaveReady'` into a static class property or a file constant. Then vf-score can import it. That way we have a single source of truth for this event name.

Ditto for the other `*Ready` events.

Reply:
And on events and names and such: I generally prefer snake-case event names, like `vf-stave-ready`.

I'd also consider using Event subclasses rather than CustomEvent, because you can fix the event options with the event class declaration. Like:

```
class StaveReadyEvent extends Event {
  static eventName = 'vf-stave-ready';

  constructor() {
    super(StaveReadyEvent.eventName, {bubbles: true});
  }
}
```

You might also see if you can use the same event for multiple elements, maybe with a name like `vf-element-ready`.